jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/CachingContentFactory.java

```diff
@@ -209,15 +209,7 @@ private CachingHttpContent(String key, String precalculatedEtag, HttpContent htt
             if (_useFileMappedBuffer)
             {
                 // map the content into memory
-                // TODO this is assuming the resource can be mapped! Inefficient to throw to test this
-                try
-                {
-                    byteBuffer = BufferUtil.toMappedBuffer(httpContent.getResource().getPath(), 0, _contentLengthValue);
-                }
-                catch (Throwable t)
-                {
-                    LOG.trace("ignored", t);
-                }
+                byteBuffer = BufferUtil.toMappedBuffer(httpContent.getResource(), 0, _contentLengthValue);
```

Comment:
Regardless of the rest of the API... the ability to avoid throwing/catching/ignoring an exception in normal handling is huge.  So I'm favourable towards this PR already.

However, for code simplicity get rid of the if above and jave a one arg version so you can just do:
```suggestion
            ByteBuffer byteBuffer = _useFileMappedBuffer ? BufferUtil.toMappedBuffer(httpContent.getResource()) : null;
```

Reply:
Done.